retrofit2-clients/src/test/java/com/palantir/remoting3/retrofit2/Retrofit2ClientBuilderTest.java

```diff
@@ -74,9 +76,30 @@ public void testUserAgent_defaultHeaderIsSent() throws InterruptedException, IOE
     }
 
     @Test
-    public void testUserAgent_invalidUserAgentThrows() throws InterruptedException {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage(is("User Agent must match pattern '[A-Za-z0-9()\\-#;/.,_\\s]+': !@"));
-        Retrofit2Client.create(TestService.class, "!@", createTestConfig("http://localhost:" + server.getPort()));
+    public void testUserAgent_usesUnknownAgentIfBogusAgentIsGiven() throws InterruptedException, IOException {
+        TestService service = Retrofit2Client.create(
+                TestService.class, "bogus user agent", createTestConfig(
+                        String.format("http://%s:%s/api/", server.getHostName().toUpperCase(), server.getPort())));
+        server.enqueue(new MockResponse().setBody("\"server\""));
```

Comment:
nit: the JaxRs tests are the same as this but look a bit nicer

Reply:
ja, haven't moved these to assertj yet. will leave for a later exercise. 
https://github.com/palantir/http-remoting/issues/602

Reply:
👍